client/src/templates/Challenges/exam/show.tsx

```diff
@@ -485,14 +483,28 @@ class ShowExam extends Component<ShowExamProps, ShowExamState> {
       showResults
     } = this.state;
 
+    const incompletePrequisites = [];
+    let qualifiedForExam = true;
+
+    for (let i = 0; i < prerequisites.length; i++) {
+      const prerequisiteChallenge = prerequisites[i];
+      const isPrerequisiteCompleted = completedChallenges.find(
+        (completedChallenge: CompletedChallenge) =>
+          prerequisiteChallenge.id === completedChallenge.id
+      );
+
+      if (!isPrerequisiteCompleted) {
+        incompletePrequisites.push(prerequisites[i].title);
+        qualifiedForExam = false;
+      }
+    }
```

Comment:
```suggestion
    const missingPrequisites = prerequisites.filter(
      prerequisite =>
        !completedChallenges.find(({ id }) => prerequisite.id === id)
    );

    const qualifiedForExam = missingPrequisites.length === 0;
```
Now you need the list of missing prequisites, the declarative approach looks even nicer imo.